### PR TITLE
Fixed: UserAgent Parsing

### LINKED ISSUE #448: [BUG]: Mylar3 UA parsed incorrectly for stats
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Screenshot_20210821-210450

### Expected Behavior

Parse as `Mylar3`

### Steps To Reproduce

_No response_

### Environment

```markdown
- OS: 
- Prowlarr: 842
- Docker Install: Yes
- Using Reverse Proxy: No
- Browser: Chrome
```

### What branch are you running?

Nightly

### Anything else?

Negative

[AB#1490](https://dev.azure.com/Servarr/7ab38f4e-5a57-4d70-84f4-94dd9bc5d6df/_workitems/edit/1490)

### PR DESCRIPTION
**Pull request by bakerboy448**
#### Database Migration
NO

#### Description
hopefully fixes UA parsing.

New:
image
Old:
image

#### Todos
- [X]  Tests - created
- N/A Translation Keys (./src/NzbDrone.Core/Localization/Core/en.json)
- N/A [Wiki Updates](https://wiki.servarr.com)

#### Issues Fixed or Closed by this PR

* Fixes #448

### CODE REVIEW

**Comment by Qstick** [comment]
I don't think you read the whole conversation, the UA is sent as 'Mylar3/hash + url'... We just want the first 'Mylar3' instead we are getting the URL now

**Comment by bakerboy448** [comment]
redux

new:
image

first attempt:
image

original:
image

**Comment by Qstick** [comment]
Looks good, yea we probably want some tests to make sure it still works with ours as well

**Comment by bakerboy448** [comment]
> Looks good, yea we probably want some tests to make sure it still works with ours as well

happy to write the tests if someone can help getting the skeleton of one framed out (and has the user agent requests handy :D)

**Comment by Qstick** [comment]
Should be something like this

```
using FluentAssertions;
using NUnit.Framework;
using NzbDrone.Common.EnvironmentInfo;
using NzbDrone.Common.Http;
using NzbDrone.Test.Common;

namespace NzbDrone.Common.Test.Http
{
    [TestFixture]
    public class UserAgentParserFixture : TestBase<UserAgentParser>
    {
        [TestCase("UA", "Parsed"]
        public void should_parse_user_agent(string userAgent, string parsedAgent)
        {
            Subject.ParseSource(userAgent).Should().Be(parsed);
        }
    }
}
```

**Comment by bakerboy448** [comment]
ToDo
- [ ] Add proper *arr examples
- [ ] Revise tests if they fail

File: src/NzbDrone.Common.Test/Http/UserAgentParserFixture.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using NzbDrone.Common.Http;
+using NzbDrone.Test.Common;
+
+namespace NzbDrone.Common.Test.Http
+{
+    [TestFixture]
+    public class UserAgentParserFixture : TestBase<UserAgentParser>
+    {
+        [TestCase("Mylar3/3ee23rh23irqfq/13123123/http://www.github.com/mylar3/mylar3/", "Mylar3")]
+        [TestCase("Lidarr", "Lidarr")]
```
Comment (Qstick) [inline src/NzbDrone.Common.Test/Http/UserAgentParserFixture.cs:13]: The arr ones have more info than this

**Review by Qstick — CHANGES_REQUESTED** [review]

File: src/NzbDrone.Common.Test/Http/UserAgentParserFixture.cs (hunk 1 above)
Comment (bakerboy448) [reply src/NzbDrone.Common.Test/Http/UserAgentParserFixture.cs:13]: got those examples handy at all?

having issues trying to find any on my end and my prowlarr dev setup is not easily setup to test *arrs <-> prowlarr locally

**Review by Qstick — APPROVED** [review]

**Pull request merged**